dev-python/QtPy/QtPy-1.9.0-r3.ebuild

```diff
@@ -0,0 +1,84 @@
+	opengl positioning printsupport qml quick script scripttools
+	scxml sensors serialport speech sql ssl svg test testlib webchannel
+	webengine webkit websockets widgets x11extras xml xmlpatterns
+"
+
+RDEPEND="
+	app-eselect/eselect-QtPy
+	|| (
+		dev-python/PyQt5[${PYTHON_USEDEP},designer?,gles2-only?,gui?,help?,location?,multimedia?,network?,opengl?,positioning?,printsupport?,sensors?,serialport?,sql?,svg?,testlib?,webchannel?,websockets?,widgets?,x11extras?,xml(+)?,xmlpatterns?]
+		dev-python/pyside2[${PYTHON_USEDEP},designer?,gles2-only?,gui?,help?,location?,multimedia?,network?,opengl(+)?,positioning?,printsupport?,sensors?,serialport(+)?,sql?,svg?,testlib?,webchannel?,websockets?,widgets?,x11extras?,xml?,xmlpatterns?]
+	)
+
+	webengine? ( || (
+		dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
+		dev-python/pyside2[${PYTHON_USEDEP},webengine]
+	) )
+
+	qml? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},qml]
+	) )
+
+	quick? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},quick]
+	) )
+
+	declarative? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},qml,quick]
+	) )
+
+	bluetooth? ( dev-python/PyQt5[${PYTHON_USEDEP},bluetooth] )
+	dbus? ( dev-python/PyQt5[${PYTHON_USEDEP},dbus] )
+	examples? ( dev-python/PyQt5[${PYTHON_USEDEP},examples] )
+	networkauth? ( dev-python/PyQt5[${PYTHON_USEDEP},networkauth] )
+	ssl? ( dev-python/PyQt5[${PYTHON_USEDEP},ssl] )
+	webkit? ( dev-python/PyQt5[${PYTHON_USEDEP},webkit] )
+
+	3d? ( dev-python/pyside2[${PYTHON_USEDEP},3d] )
+	charts? ( dev-python/pyside2[${PYTHON_USEDEP},charts] )
+	concurrent? ( dev-python/pyside2[${PYTHON_USEDEP},concurrent] )
+	datavis? ( dev-python/pyside2[${PYTHON_USEDEP},datavis] )
+	scxml? ( dev-python/pyside2[${PYTHON_USEDEP},scxml] )
+	script? ( dev-python/pyside2[${PYTHON_USEDEP},script] )
+	scripttools? ( dev-python/pyside2[${PYTHON_USEDEP},scripttools] )
+	speech? ( dev-python/pyside2[${PYTHON_USEDEP},speech] )
+"
+
+BDEPEND="test? (
+	dev-python/PyQt5[${PYTHON_USEDEP},3d(+),bluetooth(+),charts(+),concurrent(+),datavis(+),dbus(+),declarative(+),designer(+),examples(+),gui(+),help(+),location(+),multimedia(+),network(+),networkauth(+),opengl(+),positioning(+),printsupport(+),script(+),scripttools(+),scxml(+),sensors(+),serialport(+),speech(+),sql(+),ssl(+),svg(+),testlib(+),webchannel(+),webengine(+),webkit(+),websockets(+),widgets(+),x11extras(+),xml(+),xmlpatterns(+)]
+	dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
+	dev-python/pyside2[${PYTHON_USEDEP},3d(+),bluetooth(+),charts(+),concurrent(+),datavis(+),dbus(+),designer(+),examples(+),gui(+),help(+),location(+),multimedia(+),network(+),opengl(+),positioning(+),printsupport(+),qml(+),quick(+),script(+),scripttools(+),scxml(+),sensors(+),serialport(+),speech(+),sql(+),svg(+),testlib(+),webchannel(+),webengine(+),webkit(+),websockets(+),widgets(+),x11extras(+),xml(+),xmlpatterns(+)]
```

Comment:
So if these are required, just depend on them so. Are you aware what `(+)` does, are there plans to enable these by upstream or something?

Reply:
(And I guess this is to prevent the explosion when the deps gets updated?)

Reply:
> Are you aware what (+) does

Yes, it assumes the flag is enabled if it is missing from the ebuild.

The QtPy test suite skips tests for Python bindings that are not installed, this is my attempt to make sure that all of them are installed and that therefore all tests are run. At the moment some bindings are installed unconditionally by either PySide2 or PyQt5. For example, [PyQt5](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-python/PyQt5/PyQt5-5.15.2.ebuild) unconditionally installs the bindings for dev-qt/qtxml , and [PySide2](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-python/pyside2/pyside2-5.15.1.ebuild) unconditionally installs the bindings for dev-qt/qtserialport. What I have been trying to do here by using `(+)` is to make the whole thing future proof, e.g. if a future ebuild of PyQt5 no longer installs the xml bindings unconditionally this ebuild will still pull those in, or if PySide2 gains bindings for qtbluetooth those will be pulled in too. This ensures that the test suite will stay functional.

Reply:
> (And I guess this is to prevent the explosion when the deps gets updated?)

Kinda :P